projects/aca-content/viewer/src/lib/services/viewer.service.ts

```diff
@@ -0,0 +1,227 @@
+    }
+    return empty;
+  }
+
+  /**
+   * Retrieves a list of node identifiers for the folder and data source.
+   *
+   * @param source Data source name. Returns file ids for personal-files, libraries, favorites, shared and recent-files, otherwise returns empty.
+   * @param folderId Optional parameter containing folder node identifier for 'personal-files' and 'libraries' sources.
+   */
+  async getFileIds(source: string, folderId?: string): Promise<string[]> {
+    if (source === 'libraries') {
+      source = 'libraries-files';
+    }
+    const isClient = this.preferences.get(`${source}.sorting.mode`) === 'client';
+    const [sortKey, sortDirection, previousSortKey, previousSortDir] = this.getSortKeyDir(source);
+    let entries: Node[] | SharedLink[] = [];
+    let nodes: NodePaging | FavoritePaging | SharedLinkPaging | ResultSetPaging;
+
+    if (source === 'personal-files' || source === 'libraries-files') {
+      if (!folderId) {
+        return [];
+      }
+      const orderBy = isClient ? null : ['isFolder desc', `${sortKey} ${sortDirection}`];
+      nodes = await this.contentApi
+        .getNodeChildren(folderId, {
+          orderBy: orderBy,
+          fields: this.getFields(sortKey, previousSortKey),
+          where: '(isFile=true)'
+        })
+        .toPromise();
+    }
+
+    if (source === 'favorites') {
+      nodes = await this.contentApi
+        .getFavorites('-me-', {
+          where: '(EXISTS(target/file))',
+          fields: ['target']
+        })
+        .toPromise();
+    }
+
+    if (source === 'shared') {
+      nodes = await this.contentApi
+        .findSharedLinks({
+          fields: ['nodeId', this.getRootField(sortKey)]
+        })
+        .toPromise();
+    }
+
+    if (source === 'recent-files') {
+      const person = await this.contentApi.getPerson('-me-').toPromise();
+      const username = person.entry.id;
+      const query: SearchRequest = {
+        query: {
+          query: '*',
+          language: 'afts'
+        },
+        filterQueries: [
+          { query: `cm:modified:[NOW/DAY-30DAYS TO NOW/DAY+1DAY]` },
+          { query: `cm:modifier:${username} OR cm:creator:${username}` },
+          {
+            query: this.recentFileFilters.join(' AND ')
+          }
+        ],
+        fields: this.getFields(sortKey, previousSortKey),
+        include: ['path', 'properties', 'allowableOperations'],
+        sort: [
+          {
+            type: 'FIELD',
+            field: 'cm:modified',
+            ascending: false
+          }
+        ]
+      };
+      nodes = await this.contentApi.search(query).toPromise();
+    }
+
+    entries = nodes.list.entries.map((obj) => obj.entry.target?.file ?? obj.entry);
+    if (isClient) {
+      if (previousSortKey) {
+        this.sort(entries, previousSortKey, previousSortDir);
+      }
+      this.sort(entries, sortKey, sortDirection);
+    }
+    return nodes === undefined ? [] : entries.map((entry) => entry.id ?? entry.nodeId);
```

Comment:
No need to check if nodes is undefined because of multiple reasons: 
1. There is Intellij hint which says that it will give always false (because nodes will be never undefined): 
<img width="957" alt="image" src="https://github.com/Alfresco/alfresco-content-app/assets/115619721/397e8101-16e0-467c-8d27-d6fee752b384">

2. It can't be undefined because source can have following values: personal-files, libraries, favorites, shared and recent-files and you have 'if' for each of these values and in every 'if' you set nodes variable. 

3. You still refer to field of nodes variable (nodes.list) in 168th line so that means it is never undefined as you never get issue in that line that you refer to field of undefined. 

Hence please remove that condition which checks if nodes is undefined.